_assets/js/carousel.js

```diff
@@ -0,0 +1,335 @@
+    setSlides(index);
+
+    // If the carousel is animated, advance to the
+    // next slide after 5s
+    if (settings.startAnimated) {
+      timer = setTimeout(nextSlide, 5000);
+    }
+  }
+
+  // Function to set a slide the current slide
+  function setSlides(new_current, setFocusHere, transition, announceItemHere) {
+    // Focus, transition and announce Item are optional parameters.
+    // focus denotes if the focus should be set after the
+    // carousel advanced to slide number new_current.
+    // transition denotes if the transition is going into the
+    // next or previous direction.
+    // If announceItem is set to true, the live region’s text is changed (and announced)
+    // Here defaults are set:
+
+    setFocus = typeof setFocusHere !== 'undefined' ? setFocusHere : false;
+    transition = typeof transition !== 'undefined' ? transition : 'none';
+    const announceItem = typeof announceItemHere !== 'undefined' ? announceItemHere : false;
+
+    new_current = parseFloat(new_current);
+
+    let length = slides.length;
+    let new_next = new_current + 1;
+    let new_prev = new_current - 1;
+
+    // If the next slide number is equal to the length,
+    // the next slide should be the first one of the slides.
+    // If the previous slide number is less than 0.
+    // the previous slide is the last of the slides.
+    if (new_next === length) {
+      new_next = 0;
+    } else if (new_prev < 0) {
+      new_prev = length - 1;
+    }
+
+    // Reset slide classes
+    for (let i = slides.length - 1; i >= 0; i--) {
+      slides[i].className = 'slide';
+    }
+
+    // Add classes to the previous, next and current slide
+    slides[new_next].className = 'next slide' + (transition == 'next' ? ' in-transition' : '');
+    slides[new_next].setAttribute('aria-hidden', 'true');
+
+    slides[new_prev].className = 'prev slide' + (transition == 'prev' ? ' in-transition' : '');
+    slides[new_prev].setAttribute('aria-hidden', 'true');
+
+    slides[new_current].className = 'current slide';
+    slides[new_current].removeAttribute('aria-hidden');
+
+    // Update the text in the live region which is then announced by screen readers.
+    if (announceItem) {
+      carousel.querySelector('.liveregion').textContent =
+        'Item ' + (new_current + 1) + ' of ' + slides.length;
+    }
+
+    // Update the buttons in the slider navigation to match the currently displayed  item
+    if (settings.slidenav) {
+      const buttons = carousel.querySelectorAll('.slidenav button[data-slide]');
+      for (let j = buttons.length - 1; j >= 0; j--) {
+        buttons[j].className = '';
+      }
+      buttons[new_current].className = 'current';
+    }
+
+    // Set the global index to the new current value
+    index = new_current;
+  }
+
+  // Function to advance to the next slide
+  function nextSlide(announceItem) {
+    announceItem = typeof announceItem !== 'undefined' ? announceItem : false;
+
+    let length = slides.length,
```

Comment:
prefer `const` to `let` where possible
```suggestion
    const length = slides.length,
```